src/live_restore/live_restore_fs.c

```diff
@@ -919,6 +913,69 @@ __wti_live_restore_fs_fill_holes(WT_FILE_HANDLE *fh, WT_SESSION *wt_session)
     return (ret);
 }
 
+/*
+ * __wti_live_restore_cleanup_tombstones --
+ *     Remove all tombstone files from the database.
+ */
+int
+__wti_live_restore_cleanup_tombstones(WT_SESSION_IMPL *session)
+{
+    WT_DECL_RET;
+    WT_CONNECTION_IMPL *conn = S2C(session);
+    WTI_LIVE_RESTORE_FS *fs = (WTI_LIVE_RESTORE_FS *)conn->file_system;
+    WT_FILE_SYSTEM *os_fs = fs->os_file_system;
+    WT_SESSION *wt_session = (WT_SESSION *)session;
+
+    char **files;
+    uint32_t count = 0;
+    WT_DECL_ITEM(buf);
+    WT_DECL_ITEM(filepath);
+
+    fs->finished = true;
+
+    WT_RET(__wt_scr_alloc(session, 0, &filepath));
+    /*
+     * FIXME-WT-14048: Improve the logic here, we could be smart and recursively check the log
+     * directory.
+     */
+    /* Remove tombstones in the destination directory. */
+    WT_RET(os_fs->fs_directory_list(os_fs, wt_session, fs->destination.home, NULL, &files, &count));
+    for (uint32_t i = 0; i < count; i++) {
+        if (WT_SUFFIX_MATCH(files[i], WTI_LIVE_RESTORE_FS_TOMBSTONE_SUFFIX)) {
+            WT_ERR(__wt_filename_construct(
+              session, fs->destination.home, files[i], UINTMAX_MAX, UINT32_MAX, filepath));
+            __wt_verbose_info(
+              session, WT_VERB_LIVE_RESTORE, "Removing tombstone file %s", (char *)filepath->data);
+            WT_ERR(os_fs->fs_remove(os_fs, wt_session, (char *)filepath->data, 0));
+        }
+    }
+    if (F_ISSET(&conn->log_mgr, WT_LOG_CONFIG_ENABLED)) {
+        WT_ERR(__wt_scr_alloc(session, 1024, &buf));
+
+        WT_ERR(os_fs->fs_directory_list_free(os_fs, wt_session, files, count));
+        /* The log path is the only subfolder that can exist. Check its contents explicitly. */
```

Comment:
A couple nits: `subfolder` is more commonly called `subdirectory`.
The user/application can create their own subdirectories in the database home. Does MongoDB create the FTDC data directory in there? We need to make sure this works correctly with user directories existing.

Reply:
This will work correctly as it is currently implemented, @ajmorton's suggested change would actually break in the presence of the `diagnostic.data` and possible user data directories. 

This change has also been tested with a running mongodb instance with success.

Reply:
Nit fix: 5341e7207

Reply:
Then how about just changing the comment to so:
"The log path is the only WiredTiger-owned subdirectory that can exist...."

Reply:
WiredTiger-owned added in: 5362f8a43